static/app/utils/replays/hooks/useReplayData.tsx

```diff
@@ -160,33 +134,31 @@ function useReplayData({replaySlug, orgSlug}: Options): Result {
     return response.data;
   }, [api, orgSlug, projectSlug, replayId]);
 
-  const fetchSegmentList = useCallback(async () => {
-    const response = await api.requestPromise(
-      `/projects/${orgSlug}/${projectSlug}/replays/${replayId}/recording-segments/`
-    );
-    return response.data as ReplaySegment[];
-  }, [api, orgSlug, projectSlug, replayId]);
+  const fetchAllRRwebEvents = useCallback(async () => {
+    let next: ParsedHeader = {
+      href: `/projects/${orgSlug}/${projectSlug}/replays/${replayId}/recording-segments/?download&per_page=5`,
+      results: true,
+      cursor: '',
+    };
 
-  const fetchRRWebEvents = useCallback(
-    async (segmentIds: number[]) => {
-      const attachments = await Promise.all(
-        segmentIds.map(async segmentId => {
-          const response = await api.requestPromise(
-            `/projects/${orgSlug}/${projectSlug}/replays/${replayId}/recording-segments/${segmentId}/?download`,
-            {
-              includeAllArgs: true,
-            }
-          );
-
-          return decompressSegmentData(...response);
-        })
-      );
-
-      // ReplayAttachment[] => ReplayAttachment (merge each key of ReplayAttachment)
-      return flattenListOfObjects(attachments);
-    },
-    [api, replayId, orgSlug, projectSlug]
-  );
+    const segmentRanges: any = [];
+    while (next.results) {
+      const url = stripHostFromHref(next.href);
```

Comment:
out of curiosity, why do you have to do this?

Reply:
in dev i was getting urls back that had `sentry.io` as the origin/host. If i just passed them straight into `api.requestPromise` cors issues would yell at me.

I should add a comment about it, or if there are other ideas to fix 👯

Reply:
I looked at how other pagination stuff works, and that code just pulls the `cursor` value out and re-builds the url. I'll do that here instead of dropping a comment because it'll look less sketch.